src/utils_dns.c

```diff
@@ -749,22 +749,41 @@ const char *qtype_str(int t)
 	    case ns_t_srv:      return ("SRV");
 	    case ns_t_atma:     return ("ATMA");
 	    case ns_t_naptr:    return ("NAPTR");
+	    case ns_t_opt:      return ("OPT");
+# if __NAMESER >= 19991006
 	    case ns_t_kx:       return ("KX");
 	    case ns_t_cert:     return ("CERT");
 	    case ns_t_a6:       return ("A6");
 	    case ns_t_dname:    return ("DNAME");
 	    case ns_t_sink:     return ("SINK");
-	    case ns_t_opt:      return ("OPT");
-# if __NAMESER >= 19991006
+	    case ns_t_apl:      return ("APL");
 	    case ns_t_tsig:     return ("TSIG");
 # endif
+# if __NAMESER >= 20090302
+	    case ns_t_ds:       return ("DS");
+	    case ns_t_sshfp:    return ("SSHFP");
+	    case ns_t_ipseckey: return ("IPSECKEY");
+	    case ns_t_rrsig:    return ("RRSIG");
+	    case ns_t_nsec:     return ("NSEC");
+	    case ns_t_dnskey:   return ("DNSKEY");
+	    case ns_t_dhcid:    return ("DHCID");
+	    case ns_t_nsec3:    return ("NSEC3");
+	    case ns_t_nsec3param: return ("NSEC3PARAM");
+	    case ns_t_hip:      return ("HIP");
+	    case ns_t_spf:      return ("SPF");
 	    case ns_t_ixfr:     return ("IXFR");
+# endif
 	    case ns_t_axfr:     return ("AXFR");
 	    case ns_t_mailb:    return ("MAILB");
 	    case ns_t_maila:    return ("MAILA");
 	    case ns_t_any:      return ("ANY");
+# if __NAMESER >= 19991006
```

Comment:
I thought about that, but I kept the same ordering as nameser.h (ordered by type id). This will make it easier to compare on feature updates.